src/components/panels/request/response/ResponseBody.tsx

```diff
@@ -5,21 +5,27 @@ import { SprocketEditor } from '../../../shared/input/SprocketEditor';
 
 export function ResponseBody({ response }: { response: EndpointResponse }) {
 	let editorType = 'text';
-	if (response.bodyType?.toLowerCase()?.includes('json')) {
-		editorType = 'json';
-	} else if (response.bodyType?.toLowerCase()?.includes('html')) {
-		editorType = 'html';
+	const otherOptions = ['json', 'html', 'xml'];
+	for (const option of otherOptions) {
+		if (response.bodyType?.toLowerCase()?.includes(option)) {
+			editorType = option;
+			break;
+		}
 	}
-
 	return (
 		<SprocketEditor
 			ActionBarItems={
-				<Typography>
-					{response.statusCode}: {statusCodes[response.statusCode]}
-				</Typography>
+				<>
+					{response.statusCode != 0 && (
+						<Typography>
+							{response.statusCode}: {statusCodes[response.statusCode]}
+						</Typography>
+					)}
+				</>
 			}
 			height={'45vh'}
-			value={response.body}
+			// this is to maintain backwards compatibility with the old way we stored response bodies
```

Comment:
If maintaining backwards compatibility is important at this step (it will be in the future, but idk if it is now) then I think it would be a _lot_ better to just bite the bullet and make an auto-transformer that runs on workspace/file load and transforms everything into the expected/modern format, rather than scattering things throughout the code that will be a lot harder to keep up with.

Reply:
I can work on it today if you'd like, too, if you'd like to just get this in.

Reply:
sounds good